src/components/subscriptions/edit/AddSubAddon.js

```diff
@@ -0,0 +1,127 @@
+                announce({ text: t("subscriptionUpdated") });
+                onClose();
+                queryClient.invalidateQueries(SUBSCRIPTION_ADDONS_QUERY_KEY);
+            },
+            onError: setAddSubAddonError,
+        }
+    );
+
+    const handleSubmit = (values) => {
+        addSubAddon(formatSubAddonSubmission(values));
+    };
+
+    const resetState = () => {
+        setAddSubAddonError(null);
+    };
+
+    const onCloseForm = () => {
+        onClose();
+        resetState();
+    };
+
+    return (
+        <Formik
+            initialValues={{ addon_uuid: "" }}
+            onSubmit={(values, { resetForm }) => {
+                handleSubmit(values);
+                resetForm();
+            }}
+            enableReinitialize={true}
+        >
+            {({ handleSubmit, resetForm }) => (
+                <Form>
+                    <DEDialog
+                        id={parentId}
+                        open={open}
+                        onClose={() => {
+                            onCloseForm();
+                            resetForm();
+                        }}
+                        fullWidth={true}
+                        title={t("addAddon")}
+                        actions={
+                            <>
+                                <Button
+                                    id={buildID(parentId, ids.CANCEL_BUTTON)}
+                                    onClick={() => {
+                                        onCloseForm();
+                                        resetForm();
+                                    }}
+                                >
+                                    {t("cancel")}
+                                </Button>
+                                <Button
+                                    id={buildID(parentId, ids.SAVE_BUTTON)}
+                                    type="submit"
+                                    color="primary"
+                                    onClick={handleSubmit}
+                                >
+                                    {t("submit")}
+                                </Button>
+                            </>
+                        }
+                    >
+                        <Typography>{t("chooseAddon")}</Typography>
+                        <Field
+                            name="addon_uuid"
+                            component={FormTextField}
+                            variant="outlined"
+                            select
+                            size="small"
+                            id={ids.SUB_ADDONS.UUID_FIELD}
+                            label={t("addonName")}
+                        >
+                            {availableAddons?.map((addon, index) => (
+                                <MenuItem
+                                    id={ids.SUB_ADDONS.AVAILABLE_ADDONS}
+                                    key={index}
+                                    value={addon.uuid}
```

Comment:
A lot of React tutorials say it's fine to use a `map` index as the `key`, since it just has to be some unique value, but if you already have a UUID available, I think it's generally best to use that instead.

Probably not an issue here, but if the code ever needs to reorganize these items, then it may be less thrash to use the UUID for the `key`; or in the case of the app editor, where the user can reorder param items, it actually caused bugs to use an index rather than a UUID.

Reply:
Thank you for catching this, and that makes sense. I will implement this for the next PR.